### ISSUE TITLE
`finetune`, the argument of BatchNormalization's `forward` is forgotten

### ISSUE THREAD

**Issue by okdshin** [chainer/chainer#8140]
Currently, when users want to do finetune for BN, they have to implement their model to be able to change `finetune` the parameter of BN's `forward` function. It is just so hard (and no one seems to do so). `finetune` is usefull and may increase model's accuracy without any cost tho.
`finetune` should be changable as well as configurations e.g. `train`. 
https://github.com/chainer/chainer/blob/master/chainer/links/normalization/batch_normalization.py#L286

**Comment by beam2d** [comment]
Thanks for the suggestion. Do you mean you want to finetune all the BN layers in the network at once using multiple mini-batches? I think the current API is not designed primarily for that usage, while I understand that it is much more practical than what the original design considered.

FYI, the background of the current design is as follows: the finetune mode of BN was originally designed to allow reproducing the results of the original BN paper https://arxiv.org/abs/1502.03167, in which each layer's statistics is finetuned one by one (not at once!) over multiple mini-batches. It is anyway difficult to implement, but at least the `finetune` argument is better than config for this particular algorithm because users anyway have to write the logic for the finetuning into the network definition. Therefore, `finetune` was left as an argument when we introduced the configuration API.

Considering this background, I think it's good to leave the `finetune` argument, but I now also think it's great to add config entry for that as well. Argument would still be used to override the config (when it is easier to use).
